core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -58,16 +62,22 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     case UNION:
       if (!((SqlSetOperator) call.getOperator()).isAll()) {
         SqlSyntax.BINARY.unparse(writer, UNION_DISTINCT, call, leftPrec, rightPrec);
+      } else {
+        super.unparseCall(writer, call, leftPrec, rightPrec);
       }
       break;
     case EXCEPT:
       if (!((SqlSetOperator) call.getOperator()).isAll()) {
         SqlSyntax.BINARY.unparse(writer, EXCEPT_DISTINCT, call, leftPrec, rightPrec);
+      } else {
+        super.unparseCall(writer, call, leftPrec, rightPrec);
```

Comment:
Is there any test cases for this change?

Reply:
Added test case for union all and made code changes for intersect and except .